### PR TITLE
Banners Leak Protection

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Maybe fix for https://github.com/OpenRCT2/OpenRCT2/issues/11523
Unsure how often players actually delete banners with the tile inspector.

### CODE REVIEW

File: src/openrct2/world/TileInspector.cpp [hunk 1]
```diff
@@ -157,6 +203,22 @@ GameActionResult::Ptr tile_inspector_remove_element_at(const CoordsXY& loc, int1
         {
             return std::make_unique<GameActionResult>(GA_ERROR::UNKNOWN, STR_NONE);
         }
+
+        auto largeScenery = tileElement->AsLargeScenery();
+        if (largeScenery)
+        {
+            // Only delete the banner entry if there are no other parts of the large scenery to delete
+            if (numLargeScenerySequences(loc, largeScenery) == 1)
```
Comment (HMorsinkGD) [inline src/openrct2/world/TileInspector.cpp:211]: Can't test right now, but wouldn't this return 1 and thus remove the banner entry for large banner elements that have been rotated using the tile inspector?

Comment (Broxzier) [inline src/openrct2/world/TileInspector.cpp:211]: Can't test right now, but wouldn't this return 1 and thus remove the banner entry for large banner elements that have been rotated using the tile inspector?

Comment (duncanspumpkin) [reply src/openrct2/world/TileInspector.cpp:211]: Hmm never thought someone would do that.

Comment (duncanspumpkin) [reply src/openrct2/world/TileInspector.cpp:211]: What do you propose as a solution. Should I check all 4 different rotations for the scenery?

Comment (duncanspumpkin) [reply src/openrct2/world/TileInspector.cpp:211]: Ah well thats easy large scenery can't rotate!

Comment (Broxzier) [reply src/openrct2/world/TileInspector.cpp:211]: > What do you propose as a solution. Should I check all 4 different rotations for the scenery?

Yeah, that'd be enough for 2x1 banners. If we later add new banner elements, that'd require another solution. If iterating over all elements on the map is not too expensive, that might be the best thing to do here.

> Ah well thats easy large scenery can't rotate!

Ah, that does sound like a missing feature. For the Goliath letters that would be useful, for instance.

**Pull request merged**